.goreleaser.yml

```diff
@@ -16,11 +16,17 @@ snapshot:
   name_template: "{{ .Tag }}-snapshot"
 
 dockers:
+# The `latest` is not touched for pre release.
+# The template string `{{ if .Prerelease }}v{{ .Version }}{{ else }}latest{{ end }}`
+# will compute to `latest` for regular release. For pre releases, it will
+# compute to `v{{ .Version }}`. This duplicates one of the other list items
+# and therefore eliminates `latest` from the list. This is OK to do since
+# `goreleaser` builds the image only once and tagging is a cheap operation.
 - image_templates:
-  - "docker.io/vshn/k8up:latest"
+  - "docker.io/vshn/k8up:{{ if .Prerelease }}v{{ .Version }}{{ else }}latest{{ end }}"
   - "docker.io/vshn/k8up:v{{ .Version }}"
   - "docker.io/vshn/k8up:v{{ .Major }}"
-  - "quay.io/vshn/k8up:latest"
+  - "quay.io/vshn/k8up:{{ if .Prerelease }}v{{ .Version }}{{ else }}latest{{ end }}"
   - "quay.io/vshn/k8up:v{{ .Version }}"
   - "quay.io/vshn/k8up:v{{ .Major }}"
```

Comment:
We also need to prevent a push to the floating Major version tag. In other words, only the `.Version` image should be pushed

Reply:
What about `v{{ .Major }}{{ if .Prerelease }}-{{ .Prerelease }}{{ end }}`?

Reply:
that would print something like `vshn/k8up:v1-rc1` with `v1.0.0-rc1` (just tested locally with goreleaser without publishing).
so, that means any rc1 releases are floating, e.g. `1.1-rc1`, `1.2-rc1`, `1.3-rc1`... I doubt that's useful

Reply:
Agree, that is not useful.